index.js

```diff
@@ -1,18 +1,47 @@
 const isObject = require('isobject');
 
+function parse(query) {
+  if (isObject(query)) {
+    return query;
+  }
+  try {
+    return JSON.parse(
+      query
+        .replace(/(?:\r\n|\r|\n)|,|:/g, ' ')
+        .replace(/([{}])/g, ' $1 ')
+        .replace(/ +/g, ' ')
+        .trim()
+        .replace(/([^ {}]+) /g, '"$1" ')
+        .replace(/" "/g, '":true, "')
+        .replace(/} "/g, '}, "')
+        .replace(/" {/g, '": {')
+        .replace(/" }/g, '":true }')
+    );
+  } catch (error) {
+    return null;
+  }
+}
+
 module.exports = function objectql(source, query) {
-  if ((!Array.isArray(source) && !isObject(source)) || !isObject(query)) {
+  if (!Array.isArray(source) && !isObject(source)) {
+    return source;
+  }
+
+  // eslint-disable-next-line no-param-reassign
+  query = parse(query);
```

Comment:
Let's store this in a separate const and remove the `eslint-disable-next-line no-param-reassign`. Naming things is tricky, what about:

```js
const parsedQuery = parse(query);
```